commands/command_checkout.go

```diff
@@ -0,0 +1,118 @@
+package commands
+
+import (
+	"github.com/github/git-lfs/git"
+	"github.com/github/git-lfs/lfs"
+	"github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra"
+	"os"
+	"os/exec"
+	"sync"
+)
+
+var (
+	checkoutCmd = &cobra.Command{
+		Use:   "checkout",
+		Short: "Checks out LFS files into the working copy",
+		Run:   checkoutCommand,
+	}
+)
+
+func checkoutCommand(cmd *cobra.Command, args []string) {
+
+	// Parameters are filters
+	checkoutWithIncludeExclude(args, nil)
+}
+
+func init() {
+	RootCmd.AddCommand(checkoutCmd)
+}
+
+func checkoutWithIncludeExclude(include []string, exclude []string) {
+	ref, err := git.CurrentRef()
+	if err != nil {
+		Panic(err, "Could not checkout")
+	}
+
+	pointers, err := lfs.ScanTree(ref)
+	if err != nil {
+		Panic(err, "Could not scan for Git LFS files")
+	}
+
+	var wait sync.WaitGroup
+	wait.Add(1)
+
+	c := make(chan *lfs.WrappedPointer)
+
+	checkoutWithChan(c, &wait)
+	for _, pointer := range pointers {
+		if lfs.FilenamePassesIncludeExcludeFilter(pointer.Name, include, exclude) {
+			c <- pointer
+		}
+
+	}
+	close(c)
+	wait.Wait()
+
+}
+
+func checkoutAll() {
+	checkoutWithIncludeExclude(nil, nil)
+}
+
+// Populate the working copy with the real content of objects where the file is
+// either missing, or contains a matching pointer placeholder, from a list of pointers.
+// If the file exists but has other content it is left alone
+// returns immediately but a goroutine listens on the in channel for objects
+// calls wait.Done() when the final item after the channel is closed is done
+func checkoutWithChan(in <-chan *lfs.WrappedPointer, wait *sync.WaitGroup) {
+	go func() {
```

Comment:
Yeah, I was just following the pattern that had been used elsewhere (in places like catFileBatch) but I realise now that's because there's some extra startup code before the grouting starts, where in this case it's just a goroutine. I'll change it.